### PR TITLE
Add Trusty platform support

### PR DESCRIPTION
**Pull request by randomPoison**
This PR adds support for the [Trusty secure operating system](https://source.android.com/docs/security/features/trusty). This upstreams [the patch that we have been using](https://cs.android.com/android/platform/superproject/+/master:external/rust/crates/libc/patches/trusty.patch;l=1;drc=122e586e93a534160230dc10ae3474cf31dd8f7f) internally, cleaned up to meet the project's style conventions.

### CODE REVIEW

**Comment by JohnTitor** [comment]
For maintenance reasons, I'd like to add target support when it's official (the official targets have target maintainers and we could ensure whether a target is still active or not). At a glance, there's no Trusty-related target on rust-lang/rust repo, could you propose adding it there first?

**Comment by randomPoison** [comment]
I've put up a PR to add the trusty compilation targets: https://github.com/rust-lang/rust/pull/103895

Is that enough to move this PR forward? Or did you want to wait until the new targets are approved in the main Rust project before merging here? Because that PR depends on the new APIs added in this PR, which compilates things 😅

File: src/lib.rs [hunk 1]
```diff
@@ -139,6 +139,12 @@ cfg_if! {
 
         mod hermit;
         pub use hermit::*;
+    } else if #[cfg(target_os = "trusty")] {
+        mod fixed_width_ints;
+        pub use fixed_width_ints::*;
+
+        mod trusty;
+        pub use trusty::*;
```
Comment (JohnTitor) [inline src/lib.rs:147]: I wonder if we could move the module under linux_like or somewhere else similar to that. I'm not familiar with Trusty but is it completely different from Linux-like OS (i.e. exposing linux-like items on Trusty is harmful)?

**Comment by JohnTitor** [comment]
I see https://github.com/rust-lang/compiler-team/issues/582 is approved and it's a good time to go ahead.
Could you check my above comment and rebase to re-run CI?

**Comment by JohnTitor** [comment]
Closing as inactive, feel free to resolve the above comment and re-submit a PR if you're still interested in this change. Thanks for your contribution anyway!

**Pull request closed without merging**